Makefile

```diff
@@ -2092,6 +2092,7 @@ BASIC_CFLAGS += -DPAGER_ENV='$(PAGER_ENV_CQ_SQ)'
 
 ALL_CFLAGS += $(BASIC_CFLAGS)
 ALL_LDFLAGS += $(BASIC_LDFLAGS)
+ALL_LDFLAGS_WINMAIN += $(BASIC_LDFLAGS_WINMAIN)
```

Comment:
Could we use _just_ `BASIC_LDFLAGS_WINMAIN` and skip `ALL_*`? Oh, and maybe call it `BASIC_LDFLAGS_GUI_MODE` or something like that?

Reply:
@dscho  I think I found a better solution to this.  I changed things in config.mak.uname so that BASIC_LDFLAGS only contains the non-gui/console options.  Then created `CONSOLE_LDFLAGS` and `GUI_LDFLAGS`.  Then I updated the way `ALL_LDFLAGS` is built so that I can get `ALL_LDFLAGS_GUI` immediately-set before the console args are appended.  This lets the GUI version inherit the built-up flags (as well as the command line LDFLAGS (which were added to ALL_LDFLAGS way earlier)).

Similarly, I did the same for ALL_CFLAGS -- this helped resolve a compiler warning by letting me move the `-fno-stack-protector` arg back to the GCC section.

I think this model will make it much easier to add a second GUI app if we need to.